mlos_core/mlos_core/tests/__init__.py

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""
+Common functions for mlos_core Optimizer tests.
+"""
+
+from typing import List, Set, Type, TypeVar
+
+
+T = TypeVar('T')
+
+
+def _get_all_subclasses(cls: Type[T]) -> Set[Type[T]]:
+    """
+    Gets the set of all of the subclasses of the given class.
+    Useful for dynamically enumerating expected test cases.
+    """
+    return set(cls.__subclasses__()).union(
+        [s for c in cls.__subclasses__() for s in _get_all_subclasses(c)])
```

Comment:
no need to make it a list - keep it as a generator (i.e., remove the `[]` around the expression)